src/components/AqiGraph/AqiGraph.tsx

```diff
@@ -0,0 +1,311 @@
+      ]);
+    } else {
+      // Include hazardous color and extra grid line
+      /* eslint-disable-next-line no-magic-numbers */
+      setYAxisTicks([0, 50, 100, 150, 200, 300, yAxisLimit]);
+      setHorizontalFill([
+        '#7E0224',
+        '#7E0224',
+        '#8F3F97',
+        '#FF0202',
+        '#FF7E02',
+        '#FEFF00',
+        '#08E400',
+      ]);
+    }
+  }, [yAxisLimit]);
+
+  const formatLabels = (hoursAgo: number): string => {
+    const weekdays = [
+      t('Monday'),
+      t('Tuesday'),
+      t('Wednesday'),
+      t('Thursday'),
+      t('Friday'),
+      t('Saturday'),
+      t('Sunday'),
+    ];
+    const hoursPerPeriod = 12;
+    const sunday = 6;
+    // Get the local time from the user's browser
+    const date = new Date();
+    let day = date.getDay() - 1; // Subtract 1 to put in [0,6] range
+    let hour = date.getHours() - hoursAgo;
+    const minutes = date.getMinutes();
+    // Adjust values if the label is from the previous day
+    if (hour < 0) {
+      if (day === 0) {
+        day = sunday;
+      } else {
+        day -= 1;
+      }
+      hour = hoursPerDay + hour;
+    }
+    // Update AM or PM
+    let period = '';
+    if (hour > hoursPerPeriod) {
+      hour = hour % hoursPerPeriod;
+      period = ' PM';
+    } else {
+      period = ' AM';
+    }
+
+    // Formatting
+    let minutesString = '' + minutes;
+    // Convert hour 0 to 12
+    if (hour === 0) {
+      hour = hoursPerPeriod;
+    }
+    // Add leading 0 to single-digit minutes
+    /* eslint-disable-next-line no-magic-numbers */
+    if (minutes < 10) {
+      minutesString = '0' + minutes;
+    }
+    return weekdays[day] + ' ' + hour + ':' + minutesString + period;
+  };
+
+  if (sensorDocId) {
+    return (
+      <Flex
+        height="100%"
+        width="100%"
+        justifyContent="center"
+        align="center"
+        padding={1}
+      >
+        <ResponsiveContainer height={250} width="90%">
+          <ScatterChart>
+            <CartesianGrid horizontalFill={horizontalFill} fillOpacity={0.2} />
+            <XAxis
+              type="number"
+              dataKey="x"
+              height={70}
+              name="Hours Ago"
+              tick={{dy: 25, dx: -30}}
+              /* eslint-disable-next-line no-magic-numbers */
+              ticks={[0, 6, 12, 18, 24]}
+              tickFormatter={tick => formatLabels(tick)}
+              angle={-40}
+              reversed={true}
+              padding={{left: 10, right: 10}}
+              interval={0}
+              domain={[0, hoursPerDay]}
+            />
+            <YAxis
+              dataKey="y"
+              name="AQI"
+              unit=""
+              label={{value: 'AQI', position: 'Right', dx: -20, rotate: 0}}
+              ticks={yAxisTicks}
+              interval={0}
+              padding={{top: 1}}
+              domain={[0, yAxisLimit]}
+            />
+            <Scatter name="Good" data={data.good} fill="#08E400" />
```

Comment:
Translate these

Reply:
Do these appear on the screen? If not, I don't think it's necessary, but if so, I think these are already translated in a different namespace and you can just use that namespace in addition

Reply:
Although I would lower case if they don't